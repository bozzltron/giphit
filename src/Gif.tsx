import React from 'react';
import './App.css';
import Modal from 'react-modal';
import { useDispatch } from 'react-redux'
import { showOriginal } from './redux/ui/actions'
import closeIcon from './close.svg';

type GifProps = {
  id: string,
  title: string,
  preview: string,
  original: string,
  showOriginalId: string
}

Modal.setAppElement('#root')

function Gif({id, preview, original, title, showOriginalId}: GifProps) {
  const dispatch = useDispatch();

  return (
    <div className="item" onClick={()=>{ dispatch(showOriginal(id)); }}>
      <img src={preview} alt={title} />
      <Modal isOpen={id === showOriginalId}>
        <img src={original} alt={title} /> <br />
        <button onClick={(e)=>{ e.stopPropagation(); dispatch(showOriginal("")); }}>
          <img className="closeIcon" src={closeIcon} alt="close" />
        </button>
      </Modal>
    </div> 
  );
}

export default Gif;
