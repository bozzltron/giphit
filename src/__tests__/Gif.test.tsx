import React from 'react'
import Gif from '../Gif'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const initialState = {gifs: {isFetching: false, data: []}, ui: {showOriginalId:''}}
  const mockStore = configureStore()
  let store = mockStore(initialState);

  const tree = renderer
    .create(<Provider store={store}><Gif id="anyId" preview="preview.img" original="original" title="title" showOriginalId="otherId" /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});