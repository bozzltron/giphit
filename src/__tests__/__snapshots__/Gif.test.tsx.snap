// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="item"
  onClick={[Function]}
>
  <img
    alt="title"
    src="preview.img"
  />
  <div />
</div>
`;
